src/Compilers/Shared/NamedPipeUtil.cs

```diff
@@ -77,12 +77,12 @@ internal static bool CheckClientElevationMatches(NamedPipeServerStream pipeStrea
         /// <summary>
         /// Create a server for the current user only
         /// </summary>
-        internal static NamedPipeServerStream CreateServer(string pipeName)
+        internal static NamedPipeServerStream CreateServer(string pipeName, PipeDirection? pipeDirection = null)
         {
             var pipeOptions = PipeOptions.Asynchronous | PipeOptions.WriteThrough;
             return CreateServer(
                 pipeName,
-                PipeDirection.InOut,
+                pipeDirection ?? PipeDirection.InOut,
```

Comment:
seems weird to use a nullable value type instead of using PipeDirection.InOut as the default value..?

Reply:
That's the general pattern I've used for defaults in the server. Didn't want to break the trend here.